src/main/resources/apps/pantheon/modules/all/search.html

```diff
@@ -0,0 +1,111 @@
+<html>
+<head data-sly-include="${'/apps/pantheon/partials/header.html'}">
+</head>
+<body>
+
+<div class="container-fluid" data-sly-use.moduleData="com.redhat.pantheon.use.ModuleData">
+    <h1>All Modules</h1>
+
+    <p>
+        <a class="btn btn-primary" href="/modules.add.html">New Module</a>
+    </p>
+
+    <p>
+    <input type="text" id="searchTerms" onkeyup="search()" placeholder="Search for names..">
+    </p>
+
+    <table id="myTable">
+        <tr>
+            <th width="200px"><a href="#" onclick="sortTable(0)">Name</a></th>
+            <th width="325px"><a href="#" onclick="sortTable(1)">Description</th>
+            <th><a href="#" onclick="sortTable(2)">Upload Time</a></th>
+        </tr>
+
+        <sly data-sly-use.resultList="${'com.redhat.pantheon.use.DataRetriever' @ query=request.requestParameterMap['search'][0].toString}">
+        <sly data-sly-list=${resultList.data}>
```

Comment:
Can we please get rid of search.html and put all the logic back into html.html? I remember you told me that the reason for search.html was so that you could guarantee that the 'search' parameter was specified. I came up with another technique that mitigates that need:
`        <sly data-sly-test.searchIsNull="${!request.requestParameterMap['search']}"></sly>
        <sly data-sly-use.resultList="${'com.redhat.pantheon.use.DataRetriever' @ query=(searchIsNull ? '*' : request.requestParameterMap['search'][0].toString )}">`

Reply:
thanks, this is much nicer.